### PR TITLE
docs: dropdown

### PR DESCRIPTION
**Pull request by m0ksem**
While I was making docs found a lot of issues... Fixed them here.

closes https://github.com/epicmaxco/vuestic-ui/issues/1494
closes https://github.com/epicmaxco/vuestic-ui/issues/1969
closes https://github.com/epicmaxco/vuestic-ui/issues/1154
closes https://github.com/epicmaxco/vuestic-ui/issues/890
closes https://github.com/epicmaxco/vuestic-ui/issues/849
closes https://github.com/epicmaxco/vuestic-ui/issues/2125

image

### CODE REVIEW

File: packages/docs/package.json [hunk 1]
```diff
@@ -68,6 +68,7 @@
     "stylelint": "^13.13.1",
     "stylelint-config-standard": "^22.0.0",
     "ts-jest": "^26.5.4",
+    "ts-node": "^10.9.1",
```
Comment (m0ksem) [inline packages/docs/package.json:71]: We need ts-node for plop to generate docs pages.

File: packages/ui/src/components/va-input/components/VaInputWrapper/_variables.scss [hunk 2]
```diff
@@ -3,7 +3,7 @@
   --va-input-wrapper-background: var(--va-background, #f5f9fb);
   --va-input-wrapper-background-opacity: 1;
   --va-input-wrapper-border-color: var(--va-input-bordered-color);
-  --va-input-wrapper-min-width: 280px;
+  --va-input-wrapper-min-width: 240px;
```
Comment (m0ksem) [inline packages/ui/src/components/va-input/components/VaInputWrapper/_variables.scss:6]: @rustem-nasyrov . I changed min width to 240px so it perfectly fit DatePicker in DateInput.

Do you see anything wrong here?

File: packages/ui/src/composables/useMousePosition.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { useClientOnly } from './useClientOnly'
+import { onMounted, onBeforeUnmount, Ref, reactive } from 'vue'
+
+export const useMousePosition = (el?: Ref<GlobalEventHandlers | undefined | null>) => {
+  if (!el) {
+    el = useClientOnly(() => window as GlobalEventHandlers)
```
Comment (m0ksem) [inline packages/ui/src/composables/useMousePosition.ts:6]: This part can be changed to useWindow.

File: packages/ui/src/composables/useMousePosition.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  const mouse = reactive({ x: 0, y: 0 })
+
+  const onMouseMove = (e: MouseEvent) => {
+    mouse.x = e.clientX
+    mouse.y = e.clientY
+  }
+
+  onMounted(() => {
+    el?.value?.addEventListener('mousemove', onMouseMove)
```
Comment (m0ksem) [inline packages/ui/src/composables/useMousePosition.ts:17]: Better watch on target if it is not undefined.

File: packages/ui/src/components/va-input/components/VaInputWrapper/_variables.scss (hunk 2 above)
Comment (rustem-nasyrov) [reply packages/ui/src/components/va-input/components/VaInputWrapper/_variables.scss:6]: No, I don't.

File: packages/ui/src/composables/usePopover.ts [hunk 5]
```diff
@@ -109,14 +109,23 @@ const getAutoPlacement = (placement: Placement, coords: Coords, content: DOMRect
...
 }
 
+const findFirstRelativeParent = (el: Element | null) => {
+  while (el) {
+    if (window.getComputedStyle(el).position === 'relative') { return el }
+    el = el.parentElement
+  }
+
+  return document.body
+}
```
Comment (RVitaly1978) [inline packages/ui/src/composables/usePopover.ts:119]:
```suggestion
const findFirstRelativeParent = (documentRef: Ref<Document>, root: string) => {
  let el = documentRef.value.querySelector(root)

  while (el) {
    if (window.getComputedStyle(el).position === 'relative') { return el }
    el = el.parentElement
  }

  return documentRef.value.body
}
```

**Review by RVitaly1978 — CHANGES_REQUESTED** [review]
- bug (or feature) on hover: content doesn't disappear after hover:

chrome-capture-2022-6-20

- `stickToEdges` doesn't work for all components (select, time-input, date-input, popover):

https://user-images.githubusercontent.com/55198465/179948159-42da832c-6123-4849-8749-9969bcc1c3f6.mov

File: packages/ui/src/components/va-dropdown/hooks/useAnchorSelector.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  const computedAnchorRef = computed<HTMLElement | undefined>({
+    set (v) { anchorRef.value = v },
+    get () {
+      if (props.innerAnchorSelector) { return anchorRef.value?.querySelector(props.innerAnchorSelector) || anchorRef.value }
+
+      return props.anchorSelector
+        ? document.value?.querySelector(props.anchorSelector) || anchorRef.value
+        : anchorRef.value
+    },
```
[the remark this answers is not in the document]
Comment (m0ksem) [reply packages/ui/src/components/va-dropdown/hooks/useAnchorSelector.ts:22]: I believe computed things must be cached, so it doesn't run get multiple times. But, yeah, the previous idea was different, so I will refactor it.

[the remark this answers is not in the document]
Comment (m0ksem) [reply packages/ui/src/components/va-dropdown/hooks/useAnchorSelector.ts:22]: So. Computed values are actually cached until some of it's references is changed, so there will not be additional calculation on get. BUT! The important thing is that `set` [will not be recalled ](https://sfc.vuejs.org/#eNp9UcFugzAM/RUrF6hKobt2pdJu233HXDrqdnSQRE5gk1D+fU4C1bpJExKK7efn5+dJPBlTjgOKndjbhlrjwKIbzEGqtjeaHExAeC6g0b0ZHJ7Aw5l0Dxk3ZVJJ1WhlHZcH5ZCgDuh8u3oMJWZ6Cdnx2OX5CuoDTFLBgi05PSCsa3hgtC/gYbuNjQALa6te8cvNpNkzdp3OAiJVXaotyvJIzjPzcZUGwUwwT6phhDVk/K3vNcSZvgj/C7ffuomdIHVHsiB9FCqZ4we9FGb4kCJtsK+Sn+wkBw570x0dcgTSTVMS733A3WqiEMn0TX805dVqxWeJWuRcsFLsFnVS8AlCLMW7c8buqsqem3DMqy01XSp+lcRrtj2WaPvNG+lPyztfmSUuO3NUnByRNoTqhIT0H+cv6B/e6A/bI/w3vmjNOQ== )when it's dependencies updated.

It is not much important there, but if user will have a dynamic anchor prop the reactivity will be lost.

**Comment by m0ksem** [comment]
We need to make user able to remove "prevent-overflow" feature. Prevent overflow actually force us to use requestAnimationFrame which is not usually needed so we make up performance.

**Comment by m0ksem** [comment]
Dropdown must react different on events if it touch events. For example:
- trigger hover must open on touch click, since there is no hover. 
- trigger right-click must work if it is long press.

**Pull request merged**